CMakeLists.txt

```diff
@@ -115,26 +109,44 @@ if (WIN32)
         string(REPLACE "/MDd" "/MTd" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
     endif()
 else()
-    if (APPLE)
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
-    else ()
+    target_link_libraries (OpenCL PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+    if (NOT APPLE)
         set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-Wl,--version-script -Wl,${CMAKE_CURRENT_SOURCE_DIR}/loader/linux/icd_exports.map")
         if (OPENCL_ICD_LOADER_PIC)
             set_target_properties(OpenCL PROPERTIES POSITION_INDEPENDENT_CODE ON)
         endif ()
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
     endif ()
 endif ()
 
-include_directories (${OPENCL_ICD_LOADER_HEADERS_DIR})
-add_definitions (-DCL_TARGET_OPENCL_VERSION=300)
+find_package(OpenCLHeaders CONFIG NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH)
+if (TARGET OpenCL::Headers)
+  target_link_libraries (OpenCL PUBLIC OpenCL::Headers)
+else ()
+  find_path(OpenCL_HEADERS_DIR
+    NAMES
+      CL/cl.h OpenCL/cl.h
+    PATH_SUFFIXES
+      include
+      OpenCL/common/inc
+  )
+  target_include_directories(OpenCL SYSTEM PUBLIC ${OpenCL_HEADERS_DIR})
```

Comment:
I wasn't able to build the ICD loader until I built and installed the OpenCLHeaders package.  Could you please check that this configuration is supported and functional?  If this is something we'd want to support - and I think it is? - do we need to add at least one CI configuration to test it?